lib/mqtt/mqtt.cc

```diff
@@ -954,3 +1053,30 @@ int mqtt_run(Timeout *t, SObj mqttHandle)
 		  return 0;
 	  });
 }
+
+int mqtt_generate_client_id(char *buffer, size_t length)
+{
+	auto characters =
+	  std::string_view(clientIDCharacters, sizeof(clientIDCharacters) - 1);
+
+	if (length > maxClientIDLength || !length)
+	{
+		return -EINVAL;
+	}
+
+	if (!CHERI::check_pointer(buffer, length))
+	{
+		return -EINVAL;
+	}
+
+	for (size_t i = 0; i < length; i++)
+	{
+		// Select a character at random.
+		buffer[i] = characters[rand() % characters.size()];
```

Comment:
Note that, since this is not a power of two, this will give a slightly biased distribution.  It's fine for avoiding collisions, but the API should document that fact.